controllers/config.go

```diff
@@ -0,0 +1,57 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+// images allows build systems to inject images for CCCMO components
+type images struct {
+	CloudControllerManagerAWS       string `json:"cloudControllerManagerAWS"`
+	CloudControllerManagerOpenStack string `json:"cloudControllerManagerOpenStack"`
+}
+
+// operatorConfig contains configuration values for templating resources
+type operatorConfig struct {
+	ManagedNamespace string
+	ControllerImage  string
+}
+
+func getProviderFromInfrastructure(infra *configv1.Infrastructure) (configv1.PlatformType, error) {
+	if infra.Status == (configv1.InfrastructureStatus{}) || infra.Status.PlatformStatus == nil {
+		return "", fmt.Errorf("platform status is not pupulated on infrastructure")
+	}
+	if infra.Status.PlatformStatus.Type == "" {
+		return "", fmt.Errorf("no platform provider found on infrastructure")
+	}
+
+	return infra.Status.PlatformStatus.Type, nil
+}
+
+func getImagesFromJSONFile(filePath string) (images, error) {
+	var i images
+	data, err := ioutil.ReadFile(filepath.Clean(filePath))
+	if err != nil {
+		return i, err
+	}
+
+	if err := json.Unmarshal(data, &i); err != nil {
+		return i, err
+	}
+	return i, nil
```

Comment:
We should explicitly return an empty struct when there's an error, prevents side effects (eg partial unmarshal) from escaping this function
```suggestion
	data, err := ioutil.ReadFile(filepath.Clean(filePath))
	if err != nil {
		return images{}, err
	}

	var i images
	if err := json.Unmarshal(data, &i); err != nil {
		return images{}, err
	}
	return i, nil
```